packages/components/src/number-input/src/NumberInput.tsx

```diff
@@ -335,6 +331,14 @@ export function InnerNumberInput(props: InnerNumberInputProps) {
         value: inputValueRef.current
     });
 
+    useEffect(() => {
```

Comment:
I believe this is textbook what React is recommending NOT to do with `useEffect`: https://react.dev/reference/react/useEffect#caveats